bookmarks/vanilla/src/modules/Bookmark.ts

```diff
@@ -156,90 +164,81 @@ export class Bookmark {
      * @param title string
      * @param link string
      * @param session Session
-     * @returns IBookmark[]
+     * @returns IBookmark
      */
-    public static async update(url: string, title: string, link: string, session: Session) {
+    public static async update(url: string, payload: IUpdateBookmark, session: Session) {
         const indexUrl = await this.getIndexUrl(session);
         const ds = await getSolidDataset(indexUrl, { fetch: session.fetch });
-        const thing = getThing(ds, url)
+        let thing = getThing(ds, url)
 
         if (thing) {
-            let updatedBookmarkThing = buildThing(thing)
-                .setStringNoLocale(DCTERMS.title, title)
-                .setStringNoLocale(BOOKMARK.recalls, link)
-                .setUrl(RDF.type, BOOKMARK.Bookmark)
-                .build();
-
-            const updatedBookmarkList = setThing(ds, updatedBookmarkThing);
-            const updatedDataset = await saveSolidDatasetAt(indexUrl, updatedBookmarkList, { fetch: session.fetch });
-            const things = getThingAll(updatedDataset)
-
-            return things.map(thing => {
-                return {
-                    url: thing.url,
-                    title: getLiteral(thing, DCTERMS.title)?.value,
-                    link: getLiteral(thing, BOOKMARK.recalls)?.value
-                }
-            }) as IBookmark[]
+            const { title, link, creator, topic } = payload
+
+            thing = setStringNoLocale(thing, DCTERMS.title, title)
+            thing = setStringNoLocale(thing, BOOKMARK.recalls, link)
+            if (creator) thing = setNamedNode(thing, DCTERMS.creator, namedNode(creator))
+            if (topic) thing = setNamedNode(thing, BOOKMARK.hasTopic, namedNode(topic))
+            thing = setStringNoLocale(thing, "http://purl.org/dc/terms/updated", new Date().toISOString())
```

Comment:
Same as before. Let's define a constant for this scheme at the top of the file or somewhere proper.